src/OpenTK.Windowing/NativeWindow.cs

```diff
@@ -0,0 +1,477 @@
+        {
+            get
+            {
+                unsafe
+                {
+                    if (glfw.GetWindowAttrib(windowPtr, (int)WindowHint.Iconified) == 1)
+                    {
+                        return WindowState.Minimized;
+                    }
+                    else if (glfw.GetWindowMonitor(windowPtr) != null)
+                    {
+                        return WindowState.Fullscreen;
+                    }
+                    else
+                    {
+                        var monitor = glfw.GetPrimaryMonitor();
+                        var mode = glfw.GetVideoMode(monitor);
+
+                        int windowWidth, windowHeight;
+                        glfw.GetWindowSize(windowPtr, out windowWidth, out windowHeight);
+
+                        if (mode->width == windowWidth && mode->height == windowHeight)
+                        {
+                            return WindowState.Maximized;
+                        }
+                    }
+                }
+                return WindowState.Normal;
+            }
+            set
+            {
+                unsafe
+                {
+                    switch (value)
+                    {
+                        case WindowState.Normal:
+                            glfw.RestoreWindow(windowPtr);
+                            break;
+                        case WindowState.Minimized:
+                            glfw.IconifyWindow(windowPtr);
+                            break;
+                        case WindowState.Maximized:
+                            glfw.MaximizeWindow(windowPtr);
+                            break;
+                        case WindowState.Fullscreen:
+                            var monitor = glfw.GetPrimaryMonitor();
+                            var mode = glfw.GetVideoMode(monitor);
+                            glfw.SetWindowMonitor(windowPtr, monitor, 0, 0, mode->width, mode->height, mode->refreshRate);
+                            break;
+                    }
+                }
+
+                WindowStateChanged?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public WindowBorder WindowBorder
+        {
+            get
+            {
+                unsafe
+                {
+                    if (glfw.GetWindowAttrib(windowPtr, (int)WindowHint.Decorated) == 0)
+                    {
+                        return WindowBorder.Hidden;
+                    }
+                    else if (glfw.GetWindowAttrib(windowPtr, (int)WindowHint.Resizable) == 0)
+                    {
+                        return WindowBorder.Fixed;
+                    }
+
+                    return WindowBorder.Resizable;
+                }
+            }
+            set
+            {
+                throw new NotImplementedException("Requires recreating the window with different window hints https://www.glfw.org/docs/latest/window_guide.html#window_attribs");
```

Comment:
Is possible without recreation with glfwSetWindowAttrib in
https://www.glfw.org/docs/3.3/group__window.html#gace2afda29b4116ec012e410a6819033e
Version 3.3
Don't know if we want to include it though as it would need a higher version dependency(which we perhaps need anyway, as in older versions a lot of Wayland stuff is not yet implemented as well)

Reply:
Dae and I talked about it, and we decided against doing that for the time being, since GLFW 3.3 is still in development. If it's stable by the time 4.0 is ready for release, this will be implemented. The exception message has been updated to explain this.

Reply:
This has been implemented, as a check. Throws exception if the found GLFW is less than 3.3, works fine if it is. This can be closed.